src/dmd/typesem.d

```diff
@@ -2543,3 +2548,1371 @@ private extern(C++) final class ResolveVisitor : Visitor
         }
     }
 }
+
+/************************
+ * Access the members of the object e. This type is same as e.type.
+ * Params:
+ *  mt = type for which the dot expression is used
+ *  sc = instantiating scope
+ *  e = expression to convert
+ *  ident = identifier being used
+ *  flag = DotExpFlag bit flags
+ *
+ * Returns:
+ *  resulting expression with e.ident resolved
+ */
+extern(C++) Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, int flag)
```

Comment:
cc @ibuclaw does the C++ interface need this at all? If not let's make it `extern(D)` for better modularity.